File: flutter/packages/measure_flutter/test/utils/fake_id_provider.dart
import 'package:measure_flutter/src/utils/id_provider.dart';

class FakeIdProvider implements IdProvider {
  int _counter = 0;

  @override
  String spanId() {
    return 'span-id-${++_counter}';
  }

  @override
  String traceId() {
    return 'trace-id-${++_counter}';
  }

  @override
  String uuid() {
    return 'uuid-${++_counter}';
  }
}

File: android/measure-android/measure/src/test/java/sh/measure/android/fakes/FakeIdProvider.kt
package sh.measure.android.fakes

import sh.measure.android.utils.IdProvider

internal class FakeIdProvider(
    var id: String = "fake-id",
    private val autoIncrement: Boolean = false,
) : IdProvider {
    private var counter = 0
    val generatedIds = mutableListOf<String>()

    override fun uuid(): String = if (autoIncrement) {
        val generatedId = "$id-${++counter}"
        generatedIds.add(generatedId)
        generatedId
    } else {
        id
    }

    override fun spanId(): String = if (autoIncrement) "$id-span-${++counter}" else id

    override fun traceId(): String = if (autoIncrement) "$id-trace-${++counter}" else id
}
